### ISSUE TITLE
Admin interface doesn't work with manual build

### ISSUE THREAD

**Issue by Xheradon** [sulu/sulu#6031]
| Q | A
| --- | ---
| Sulu Version | 2.2.6, 2.2.7, 2.2.8
| Browser Version | Chrome 90.0.4430.93

# Actual Behavior
An error ocurrs after loging in the admin interface:

```js
Route.js:42 Uncaught (in promise) TypeError: Object(...) is not a function
    at z.get regexp (Route.js:42)
    at me (mobx.module.js:801)
    at e.computeValue (mobx.module.js:1286)
    at e.trackAndCompute (mobx.module.js:1271)
    at e.get (mobx.module.js:1231)
    at e.read (mobx.module.js:4139)
    at z.get (mobx.module.js:4347)
    at z.get (mobx.module.js:343)
    at W.match (Router.js:241)
    at Oe (mobx.module.js:951)
```

# Expected Behavior
No error.

# Steps to Reproduce
I've build the admin manually following the docs (with the command, with docker and manually). The error happens with and without custom extensions loaded and seems to be related to `path-to-regexp`, or at least with how the modules are being imported. The error wasn't there with Sulu 2.2.5.

If I import path-to-regexp like this: `import pathToRegexp from 'path-to-regexp';` the error goes away and appears later when using `import {compile} from 'path-to-regexp';`. The installed library version is v6.2.0, so there shouldn't be a problem importing like that.

The error doesn't happen if I use the pre-built admin.

**Comment by niklasnatter** [comment]
Hey,
is it possible that the `assets/admin` folder of your project is not in sync with the respective version of the skeleton (eg. the [2.2.8 tag](https://github.com/sulu/skeleton/tree/2.2.8/assets/admin))?

Please follow the [Upgrading Sulu 2.x](https://docs.sulu.io/en/2.2/upgrades/upgrade-2.x.html) guide when updating a project. The `sulu:admin:update-build` command described in the last step should verify that the `assets/admin` folder matches installed version.

**Comment by Xheradon** [comment]
Checked that already. I deleted all files except for index.js and package.json and merged those two.

**Comment by niklasnatter** [comment]
I just set up a fresh `sulu/skeleton`, added a `console.log('test');` to the `assets/admin/index.js` and built the administration interface using `bin/console sulu:admin:update-build`. The process worked for me and i see the log in the console of my browser. 

I suspect that the problem is caused by something inside of your project. Can you please provide a repository that allows us to reproduce the problem?

**Comment by Xheradon** [comment]
Funny thing, while trying to make a demo proyect to share after copying the js to the demo proyect and making the build it works. The old proyect still doesn't work, but if I copy the "demo build" to the old proyect it works.

Could you maybe tell me if there are any other files outside the admin folder and apart from the libraries installed (or symlinked) from the package.json file that affect the build process?

**Comment by niklasnatter** [comment]
> Could you maybe tell me if there are any other files outside the admin folder and apart from the libraries installed (or symlinked) from the package.json file that affect the build process?

Yes, unfortunately npm is not able to install the correct dependencies if your project contains outdated `node_modules` folders or `package-lock.json` files in some cases (in my experience this mostly happens when switching between different `sulu/sulu` minor versions). 

This is why the `sulu:admin:update-build` command cleans up these files before starting the build: https://github.com/sulu/sulu/blob/685a5245eff455480c768e8f5b5ec526d883d3da/src/Sulu/Bundle/AdminBundle/Command/UpdateBuildCommand.php#L293-L333

Have you used the `sulu:admin:update-build` command to create the build? 
If yes, can you check if there are any remaining `node_modules` folders or `package-lock.json` files in your project if you abort the command after the cleanup? (I think in modern shells you could use `ls ./**/package-lock.json` and `ls ./**/node_modules` to check this)

**Comment by Xheradon** [comment]
I tend to use `sulu:admin:update-build`, yes. A weird thing has been happening when I use it: it deletes both the `./assets/admin/node_modules` folder and `./assets/admin/package-lock.json` but it also deletes my `./vendor` folder.

```
Start manual build ...
======================

Cleanup previously installed "node_modules" folders
---------------------------------------------------

Install npm dependencies
------------------------

PHP Fatal error:  Uncaught Error: Class 'Symfony\Component\ErrorHandler\ErrorEnhancer\UndefinedFunctionErrorEnhancer' not found in .\vendor\symfony\error-handler\ErrorHandler.php:738
```

I then have to use `composer install` to get my dependencies back. After that I can use `sulu:admin:update-build` again. Seems like if I have a `node_modules` folder it deletes more than it should.

I'm on Windows if it helps.

Anyway, I also tried to make the build with docker and cleaning both the lock file and the node_modules folder manually and the error persisted.

**Comment by Xheradon** [comment]
After cloning the proyect in another folder and trying in 2 more machines I still can't replicate it. I guess some I have some corrupted folder or something.

The deletion of the `./vendor` folder still happens though. Should I open another issue with this?

I'm closing this issue, thanks for the help!
